src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs

```diff
@@ -264,15 +264,27 @@ public override uint Send(SNIPacket packet)
         /// Receive a packet synchronously
         /// </summary>
         /// <param name="packet">SNI packet</param>
-        /// <param name="timeout">Timeout</param>
+        /// <param name="timeoutInMilliseconds">Timeout in Milliseconds</param>
         /// <returns>SNI error code</returns>
-        public override uint Receive(ref SNIPacket packet, int timeout)
+        public override uint Receive(ref SNIPacket packet, int timeoutInMilliseconds)
         {
             lock (this)
             {
                 try
                 {
-                    _tcpClient.ReceiveTimeout = (timeout != 0) ? timeout : 1;
+                    if (timeoutInMilliseconds < 0)
+                    {   //  infinite timeout (-1) means (0) in TCPClient 
+                        _tcpClient.ReceiveTimeout = 0;
+                    }
+                    else if (timeoutInMilliseconds == 0)
+                    {
+                        // 0 timeout is equal -1 in TCPClient
+                        _tcpClient.ReceiveTimeout = -1;
```

Comment:
Unfortunately this is undocumented and I don't recommend using it:
* 0 means that the socket will not timeout.
* -1 isn't defined.

https://msdn.microsoft.com/en-us/library/windows/desktop/ms740532(v=vs.85).aspx

> The timeout, in milliseconds, for blocking receive calls. The default for this option is zero, which indicates that a receive operation will not time out. If a blocking receive call times out, the connection is in an indeterminate state and should be closed.